decoder/vaapidecoder_h264.cpp

```diff
@@ -1625,7 +1633,11 @@ bool VaapiDecoderH264::isDecodeContextChanged(const SharedPtr<SPS>& sps)
                                               : sps->m_width;
     uint32_t height = sps->frame_cropping_flag ? sps->m_cropRectHeight
                                                : sps->m_height;
-    if (setFormat(width, height, sps->m_width, sps->m_height, maxDecFrameBuffering + 1)) {
+    if (m_configBuffer.flag & HAS_SURFACE_NUMBER && m_configBuffer.surfaceNumber > 0)
```

Comment:
no need change this, the surface calculator has done by the setFormat

Reply:
setFormat() just use the surface number passed from format parameter.